### ISSUE TITLE
Project creation requires existing project dir

### ISSUE THREAD

**Issue by jluethi** [fractal-analytics-platform/fractal-web#48]
Currently, creating a new project via the web client requires that the project directory already exists. That's not a requirement we should have when users actually use this.
Actually, in many cases, the user probably doesn't want to think about what the project directory is, because the main use for it is to put some logs and submission files there (when running in slurm mode) and we "misuse" the project directory as the directory for the output dataset in some examples. 

This will likely require some upstream changes: On the CLI side, we create the project dir before we create a project as well and I assume that's why the project creation endpoint requires an existing directory. We should think about whether the directory can also be created at that moment (not trivial, as it needs to be a user action when the user and server aren't the same).
Also, we should think whether there would be some default base directory where project directories are created, such that the user does not need to specify anything about this directory by default. I'll open a corresponding fractal-server issue.

----

For context:
If I try to create a project with a directory that doesn't exist yet, I get:
```
There has been an error, reason:
{
  "detail": "Project dir /tmp/new_proj_dir does not exist"
}
```

**Comment by tcompa** [comment]
This discussion should continue in https://github.com/fractal-analytics-platform/fractal-server/issues/553.
Let's come back to the fractal-web issue only after upstream changes are decided and implemented.

**Comment by jluethi** [comment]
Will be solved by adopting fractal-server 1.2.0, see https://github.com/fractal-analytics-platform/fractal-web/issues/92
